store/store.go

```diff
@@ -440,24 +441,38 @@ func (bs *BlockStore) SaveBlock(block *types.Block, blockParts *types.PartSet, s
 		bs.saveBlockPart(height, i, part)
 	}
 
-	// Save block meta
-	blockMeta := types.NewBlockMeta(block, blockParts)
-	pbm := blockMeta.ToProto()
-	if pbm == nil {
-		panic("nil blockmeta")
-	}
-	metaBytes := mustEncode(pbm)
-	if err := bs.db.Set(blockMetaKey(height), metaBytes); err != nil {
-		panic(err)
-	}
-	if err := bs.db.Set(blockHashKey(hash), []byte(fmt.Sprintf("%d", height))); err != nil {
-		panic(err)
-	}
+	// Save the block meta data
+	bs.tr.Append(func(dbm.DB) {
```

Comment:
How does this work exactly? Is the `DB` argument from the function arg not used? How does the transactor relate to the underlying `DB`?

Reply:
Heres the PR for the transactor, https://github.com/tendermint/tm-db/pull/147/files within your guy's `tm-db` repo.  The `DB interface` gets passed into transactor at L475.  It utilizes `DB` inherit `Set` and `Delete` functionality when `Transactor's` `Transact` method is called.

Reply:
Currently this PR does not take into account dissimilarities between the DBs except vaguely (and incompletely) with 
https://github.com/tendermint/tm-db/pull/147/files#diff-63ab601287b8b9c040760fe0bdd288f55b73f37cd7e4f1e519bea2bd43a18bbaR104.

Also, as of now, it only tackles the A aspect of ACID, atomicity.

Reply:
Also as you pointed out correctly, it looks I should get rid of the `dbm.DB` passing variable.

Reply:
As you correctly noticed the passing of `DB` there was unnecessary.  Only needed when running `Transact` so the `Transactor's `rollback func` can utilize the `DB's` `delete` method.